pay-api/src/pay_api/services/refund.py

```diff
@@ -311,9 +311,15 @@ def _validate_partial_refund_lines(refund_revenue: List[RefundPartialLine], invo
     @classmethod
     def _validate_allow_partial_refund(cls, refund_revenue, invoice: InvoiceModel):
         if refund_revenue:
-            if invoice.payment_method_code != PaymentMethod.DIRECT_PAY.value:
+            if not cls._is_payment_method_allowed_for_partial_refund(invoice.payment_method_code):
                 raise BusinessException(Error.PARTIAL_REFUND_PAYMENT_METHOD_UNSUPPORTED)
 
+    @classmethod
+    def _is_payment_method_allowed_for_partial_refund(cls, payment_method_code: str) -> bool:
+        """Check if the payment method is allowed for partial refunds."""
+        allowed_methods = [PaymentMethod.DIRECT_PAY.value, PaymentMethod.PAD.value, PaymentMethod.ONLINE_BANKING.value]
```

Comment:
`payment_methods` table, probably a better place to keep these - might make sense to have a column for partial refunds there